### PR TITLE
Ruff ann001 + ann002

### PR DESCRIPTION
**Pull request by alexauritt**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

chips away at, but does not fully reslve: https://github.com/strawberry-graphql/strawberry/issues/2477

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by alexauritt** [comment]
@patrick91 heya - looking to start this up again, should be able to contribute a good more next few weeks.

Bit stuck on this - mypy and ruff are passing locally for me, in and out of virtual env, w and w/o the mypy.ini:

Screenshot 2023-04-01 at 12 06 42 PM

Am I running this incorrectly?

**Comment by alexauritt** [comment]
@patrick91 👋 - heya - I'm back! Hoping to start putting in more time contributing.. :)
Stuck on this one - can you advise? mypy and ruff are passing locally for me - I'm not able to repro the mypy errors I'm seeing here. (I'm running `poetry run mypy` - 

> @patrick91 heya - looking to start this up again, should be able to contribute a good more next few weeks.
> 
> ~Bit stuck on this - mypy and ruff are passing locally for me, in and out of virtual env, w and w/o the mypy.ini:~
> 
> Screenshot 2023-04-01 at 12 06 42 PM
> 
> ~Am I running this incorrectly?~

disregard - had some stale dependencies...

**Comment by botberry** [comment]
Hi, thanks for contributing to Strawberry 🍓!

We noticed that this PR is missing a `RELEASE.md` file. We use that to automatically do releases here on GitHub and, most importantly, to PyPI!

So as soon as this PR is merged, a release will be made 🚀.

Here's an example of `RELEASE.md`:

```markdown
Release type: patch

Description of the changes, ideally with some examples, if adding a new feature.
```

Release type can be one of patch, minor or major. We use [semver](https://semver.org/), so make sure to pick the appropriate type. If in doubt feel free to ask :)


Here's the tweet text: 
```
🆕 Release (next) is out! Thanks to Alex Auritt for the PR 👏

Get it here 👉 https://github.com/strawberry-graphql/strawberry/releases/tag/(next)
```

File: strawberry/auto.py [hunk 1]
```diff
@@ -24,11 +24,11 @@ class StrawberryAutoMeta(type):
 
     """
 
-    def __init__(self, *args, **kwargs):
+    def __init__(self, *args: str, **kwargs: Dict[str, Any]):
```
Comment (patrick91) [inline strawberry/auto.py:27]: I think in all kwargs we want Any (or the correct type): https://mypy-play.net/?mypy=latest&python=3.11&gist=925cfe7dea572e1a3a0de141e8bd5288

Comment (alexauritt) [reply strawberry/auto.py:27]: Will patch tonight. Thanks!

Comment (alexauritt) [reply strawberry/auto.py]: > Will patch tonight. Thanks!

patched - but now seeing a single test error I'm not able to replicate locally...

stuck on this for now

Comment (alexauritt) [reply strawberry/auto.py]: ok - not sure if it was a fluke or what - but I can't repro that previous test failure locally. Forced re-runs a couple of times be reverting and re-reverting. 🤷

Comment (alexauritt) [reply strawberry/auto.py]: Increasingly looking like that one red build was a fluke. I'll try and keep this branch up to date for easier merging but let me know if anything else needs attention. Thanks @patrick91!

**Review by patrick91 — APPROVED** [review]
Thanks @alexauritt I think this is fine, we can definitely remove some of the Anys in future PRs :)

**Pull request merged**